homeassistant/components/sensor/kwb.py

```diff
@@ -0,0 +1,129 @@
+"""
+Support for KWB Easyfire.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.kwb/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import (CONF_HOST, CONF_PORT, CONF_DEVICE,
+                                 EVENT_HOMEASSISTANT_STOP,
+                                 STATE_UNKNOWN)
+from homeassistant.helpers.entity import Entity
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+import homeassistant.helpers.config_validation as cv
+
+
+REQUIREMENTS = ['pykwb==0.0.8']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_HOST = 'localhost'
+DEFAULT_PORT = 23
+DEFAULT_TYPE = 'tcp'
+DEFAULT_RAW = False
+DEFAULT_DEVICE = '/dev/ttyUSB0'
+
+MODE_SERIAL = 0
+MODE_TCP = 1
+
+CONF_TYPE = 'type'
+CONF_RAW = 'raw'
+
+"""
+SERIAL_SCHEMA = {
+    vol.Required(CONF_DEVICE, default=DEFAULT_DEVICE): cv.string,
+    vol.Required(CONF_TYPE, default=DEFAULT_TYPE): 'serial',
+    vol.Optional(CONF_RAW, default=DEFAULT_RAW): cv.boolean,
+}
+
+ETHERNET_SCHEMA = {
+    vol.Required(CONF_HOST, default=DEFAULT_HOST): cv.string,
+    vol.Required(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    vol.Required(CONF_TYPE, default=DEFAULT_TYPE): 'tcp',
+    vol.Optional(CONF_RAW, default=DEFAULT_RAW): cv.boolean,
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    DOMAIN: vol.Any(SERIAL_SCHEMA, ETHERNET_SCHEMA)
+})
+"""
```

Comment:
ok, I would have preferred the more elaborate way to validate, but I guess the simpler way will be all right for now